bcp47/tests/Data/BCP47/TrieSpec.hs

```diff
@@ -94,6 +94,10 @@ spec = do
       let Just trie = fromList [(en, "color"), (enGB, "colour")]
       lookup enGBTJP trie `shouldBe` Just "colour"
 
+    it "lookups case-insensitively" $ do
+      let Just trie = fromList [(enTJPUpper, "color"), (enGBTJPUpper, "color")]
+      lookup enTJP trie `shouldBe` Just "color"
```

Comment:
Yeah, so I see why you made the `Upper` variants.

I do still feel that having them in the main interface is confusing, but I see the very real use-case here. WDYT of defining them in the test module instead? I think it might even be clearer to see "this thing was construstructed from lower" and "this thing was constructed from upper" right here. I admit that's what the `Upper` suffix is literally indicating, but it's not as clear to me as just seeing the construction right here. (And you know know how I'd feel with naming things `enGBConstructedFromUpperText` :stuck_out_tongue:)

That said, I personally find the `describe "Eq"` test cases to be enough for the case-insensitivity. If we're confident the current tests sufficiently prove that `lookup` is built on `Eq`, then testing case-insensitivity here would be unnecessary.

Reply:
The lookups are built on `Ord`, which is a little tougher to devise a test for. And, yes, will try to move the test values

Reply:
That is how they're being constructed in [TestImport.hs](https://github.com/freckle/bcp47/pull/26/files#diff-d591ad16863ab29ff5490a7477de9a5a26a458afb7dc02f59f24942a15c50cd8R29) now. I can inline those at the use sites if you think that's easier to read.

Reply:
>  I can inline those at the use sites if you think that's easier to read.

I do think it's clearer to see that construction of a `BCP47` value from upper-case text, than to look at a `BCP47` value with the suffix `Upper` in its name. To get the same clarity in the latter case, I think you'd have to name it `BuiltFromUpper` or some such, since a `BCP47` just can't _be_ "upper". And any name like that is just silly long, so in-lining it makes the most sense to me.

But, don't let that long-winded explanation imply I feel strongly, I don't. Approve totally stands either way.